var/spack/repos/builtin/packages/netcdf-cxx/package.py

```diff
@@ -29,8 +29,8 @@ def libs(self):
         shared = True
         return find_libraries("libnetcdf_c++", root=self.prefix, shared=shared, recursive=True)
 
-    def configure_args(self):
-        args = []
+    def cmake_args(self):
+        args = [self.define_from_variant("USE_NETCDF4", "netcdf4")]
```

Comment:
The error at: https://gitlab.spack.io/spack/spack/-/jobs/5522408 looks legitimate.

```
>> 9    CMake Error: The source directory "/tmp/root/spack-stage/spack-stage-netcdf-cxx-4.2-yax57ewy54t2cr2bvkkfgltizxtkcz5h/spack-build-yax57ew/CPPFLAGS=-DUSE_NETCDF4" does not exist.
```

It's not clear to me why this change is causing the error but when I examine the surrounding logic I can say at least that we should no longer have to set `CPPFLAGS` (in fact all of the below search logic may be unnecessary).

Reply:
I took a closer look at the CMake system for NetCDF-CXX and you're absolutely right, we don't need any of the find logic. I've taken it all out and pushed. Hopefully Gitlab CI will work now.